source/s3_request_messages.c

```diff
@@ -402,24 +402,28 @@ struct aws_http_message *aws_s3_get_source_object_size_message_new(
         goto error_cleanup;
     }
 
-    struct aws_byte_cursor source_bucket;
-    AWS_ZERO_STRUCT(source_bucket);
-    struct aws_byte_cursor source_key;
-    AWS_ZERO_STRUCT(source_key);
-
-    source_bucket.ptr = decode_buffer.buffer;
-
-    /* source format is {bucket}/{key}. split them */
-    for (size_t i = 0; i < decode_buffer.len; i++) {
-        if (decode_buffer.buffer[i] == '/') {
-            source_bucket.len = i;
-            source_key.ptr = decode_buffer.buffer + i + 1; /* skip the / */
-            source_key.len = decode_buffer.len - i - 1;
+    /* extracts source bucket and key */
+    struct aws_byte_cursor source_bucket = aws_byte_cursor_from_buf(&decode_buffer);
+
+    /* x-amz-copy-source might have an optional leading slash. if so let's skip it */
```

Comment:
Is it an error for get_object to have a leading slash? This function is also used for get_object right? Will this break the current behavior for that?

Reply:
This function is not used by get_object. It is used only by Copy meta requests to extract info from the source object specified in the `x-amz-copy-source` header. Maybe I should rename the function to `aws_s3_get_copy_source_object_size_message_new` to avoid confusion? The format of the header value can be either `/{bucket}/{key}` or `{bucket}/{key}` as described in the [docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html). Thanks for checking!

Reply:
If it's only used for copy object, that's fine. Yeah, maybe rename or update the documentation for this function to avoid the confusion. I saw the documentation saying:
```
/* Creates a HEAD GetObject request to get the size of the source object. */
```
So, I assume GetObject also uses this function.